### PR TITLE
CSS_grid_layout/basic_concept: upgrade and fix fr

### PR DESCRIPTION
**Pull request by pols12**
### Description

Upgrades French localization to include English fixes and improvements from September 2020.

### Motivation

Code samples were bugged because of their accumulation in the same section. I noted other improvements have been done in English source, so I imported them.

### Additional details

Note I have kept `code` syntax inside headings, not applying mdn/content#19336, since I think it’s better for screen-readers, especially in translations (since these terms come from English).

Also, I have been afraid by this gap between original and French l10n.
* Does a tool exist to follow updates to a page and to see outdated translations?
* Is each date update noted, in order to easier check what have changed since last update?
* Do we have a tool to provide a diff for a given file between two dates?

### CODE REVIEW

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 1]
```diff
@@ -17,7 +18,7 @@ Une grille est un ensemble de lignes horizontales et verticales qui se croisent
 
 ### Pistes à taille fixe ou variable
 
-On peut créer une grille avec des pistes à taille fixes en utilisant une unité comme le pixel. Pour les pistes à taille variable on peut utiliser le pourcentage ou la nouvelle unité `fr` créée à cet effet.
+On peut créer une grille avec des pistes à taille fixes en utilisant une unité comme le pixel. Pour les pistes à taille variable on peut utiliser le pourcentage ou l’unité `fr` créée à cet effet.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:21]:
```suggestion
On peut créer une grille avec des pistes à taille fixes en utilisant une unité comme le pixel. Pour les pistes à taille variable on peut utiliser le pourcentage ou l'unité `fr` créée à cet effet.
```

Pas d'apostrophe courbe :)

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 2]
```diff
@@ -130,9 +135,9 @@ Nous avons créé une grille avec trois pistes de 200 pixels de large. Chaque é
 }
 ```
 
-{{ EmbedLiveSample('Les_pistes', '610', '140') }}
+{{ EmbedLiveSample('Exemple_simple', '610', '140') }}
 
-### L'unité fr
+### L’unité `fr`
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:140]:
```suggestion
### L'unité `fr`
```

(apostrophe droite ici :))

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 3]
```diff
@@ -169,7 +174,9 @@ Les pistes peuvent être définies à l'aide de n'importe quelle unité de mesur
 }
 ```
 
-{{ EmbedLiveSample("L'unité_fr", '220', '140') }}
+{{ EmbedLiveSample("L’unité_fr", '220', '140') }}
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:177]:
```suggestion
{{ EmbedLiveSample("L'unité_fr", '220', '140') }}
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 4]
```diff
@@ -180,7 +187,39 @@ L'exemple suivant crée une grille avec une colonne de `2fr`, et deux colonnes d
 }
 ```
 
-Dans ce dernier exemple nous utilisons à la fois des dimensions absolues et des relatives. La première colonne faisant 500px, cette valeur est soustraite de l'espace disponible. L'espace restant est divisé en trois et alloué proportionnellement aux deux colonnes spécifiées avec l'unité relative `fr`.
+```html hidden
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:190]:
```suggestion
```html
```

cf. [le fichier en-US](https://github.com/mdn/content/blob/main/files/en-us/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md?plain=1#L196)

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 5]
```diff
@@ -189,6 +228,36 @@ Dans ce dernier exemple nous utilisons à la fois des dimensions absolues et des
 }
 ```
 
+```html hidden
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:231]:
```suggestion
```html
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 6]
```diff
@@ -377,9 +446,33 @@ Le second élément commence sur la ligne de colonne 1 et s'étend sur une seule
...
 ```
 
-{{ EmbedLiveSample('Positionnement_des_éléments_sur_les_lignes', '230', '420') }}
+{{ EmbedLiveSample('Positionnement_des_éléments_sur_les_lignes', '230', '450') }}
+
+> **Note :** Pensez à utiliser l'[Inspecteur de grille](https://firefox-source-docs.mozilla.org/devtools-user/page_inspector/how_to/examine_grid_layouts/index.html) dans les outils de développement de Firefox pour voir comment les éléments se placent sur les lignes d'une grille.
+
+### Raccourcis de positionnement sur ligne
+
+Les valeurs écrites en entier utilisées plus haut peuvent être abrégées en une ligne pour les colonnes avec {{cssxref("grid-column")}} et une ligne pour les rangées avec {{cssxref("grid-row")}}. L’exemple suivant donne le même positionnement que le code précédent, mais avec beaucoup moins de CSS. La valeur avant la barre oblique (`/`) représente la ligne de début, celle d’après représente la ligne de fin.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:455]:
```suggestion
Les valeurs écrites en entier utilisées plus haut peuvent être abrégées en une ligne pour les colonnes avec {{cssxref("grid-column")}} et une ligne pour les rangées avec {{cssxref("grid-row")}}. L'exemple suivant donne le même positionnement que le code précédent, mais avec beaucoup moins de CSS. La valeur avant la barre oblique (`/`) représente la ligne de début, celle d'après représente la ligne de fin.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 7]
```diff
@@ -377,9 +446,33 @@ Le second élément commence sur la ligne de colonne 1 et s'étend sur une seule
...
 
-Pensez à utiliser l'[Inspecteur de grille](/fr/docs/Tools/Page_Inspector/How_to/Examine_grid_layouts) dans les outils de développement pour voir comment les éléments se placent sur les lignes d'une grille items.
+Vous pouvez omettre la valeur de la ligne de fin si la zone ne recouvre qu’une piste.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:457]:
```suggestion
Vous pouvez omettre la valeur de la ligne de fin si la zone ne recouvre qu'une piste.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 8]
```diff
@@ -406,9 +499,9 @@ Les _gouttières_ entre les cellules sont définies à l'aide des propriétés {
 }
 ```
 
-> **Note :** Les anciens navigateurs utilisent {{cssxref("column-gap")}}, {{cssxref("row-gap")}}, {{cssxref("gap")}} avec le préfixe `grid-` soit : {{cssxref("grid-column-gap")}}, {{cssxref("row-gap")}} et {{cssxref("gap")}}.
+> **Note :** Les anciens navigateurs utilisent {{cssxref("column-gap")}}, {{cssxref("row-gap")}}, {{cssxref("gap")}} avec le préfixe `grid-` soit : {{cssxref("grid-column-gap")}}, {{cssxref("grid-row-gap")}} et {{cssxref("grid-gap")}}.
 >
-> Les navigateurs actuels retirent progressivement ce préfixe (la version préfixée sera maintenue sous forme d'alias). À l'heure actuelle, certains navigateurs ne prennent pas encore la version sans préfixe et c'est pourquoi certains exemples de ce guide continuent d'utiliser les versions préfixées avec `grid-`.
+> Les navigateurs prennent dorénavant tous en charge les valeurs sans ce préfixe, mais les versions préfixées continuent d’être prises en charge sous forme d’alias.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:504]:
```suggestion
> Les navigateurs prennent dorénavant tous en charge les valeurs sans ce préfixe, mais les versions préfixées continuent d'être prises en charge sous forme d'alias.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 9]
```diff
@@ -444,6 +537,12 @@ L'espace utilisé par les gouttières sera pris en compte avant l'assignation de
 
 Un élément placé dans une grille peut aussi être le conteneur d'une autre grille. Dans l'exemple suivant nous retrouvons la grille de trois colonnes créée plus haut, avec deux éléments explicitement positionnés. Le premier élément contient lui-même des éléments. Comme ils ne sont pas des enfants directs de la grille principale, ils se positionnent normalement dans le flux.
 
+![Nested grid in flow](grid3.jpg)
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:540]:
```suggestion
[Grille imbriquée dans le flux](1_nested_grids_in_flow.png)
```

Ici on veut afficher la même image que celle sur la page anglaise, donc il faut utiliser le même nom de fichier :)

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 10]
```diff
@@ -444,6 +537,12 @@ L'espace utilisé par les gouttières sera pris en compte avant l'assignation de
...
+
+### Imbrication sans sous-grille
+
+En définissant la propriété `display:grid` sur l'élément `box1`, il devient lui-même une grille et ses enfants se positionnent sur cette grille.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:544]:
```suggestion
En définissant la propriété `display: grid` sur l'élément `box1`, il devient lui-même une grille et ses enfants se positionnent sur cette grille.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 11]
```diff
@@ -495,15 +598,17 @@ En définissant la propriété `display:grid` sur l'élément `box1`, il devient
...
 
 Dans ce cas la grille imbriquée n'est pas liée à la grille qui la contient. Elle n'hérite pas des {{cssxref("gap")}}, et ses lignes ne s'alignent pas avec celles de la grille parent.
 
 ### Sous-grille
 
-Dans le brouillon de travaille pour la spécification de niveau 2 pour CSS Grid, il existe une fonctionnalité nommée _sous-grille_ qui permet de créer des grilles imbriquées qui utilisent la définition de la grille parent. _Ceci n'est implémenté dans aucun navigateur pour le moment, et la spécification peut encore changer._ Pour l'utiliser telle qu'elle est définie actuellement il faudrait modifier l'exemple suivant et remplacer `display: grid` par `display: subgrid`, et supprimer la définition des pistes. La piste imbriquée utiliserait les pistes de la grille parent pour positionner ses éléments.
+Dans le brouillon de travail pour la spécification de niveau 2 pour CSS Grid, il existe une fonctionnalité nommée _sous-grille_ qui permet de créer des grilles imbriquées qui utilisent la définition de la grille parent.
+
+> **Note :** Cette fonctionnalité apparait dans Firefox 71, le seul navigateur pour le moment à la prendre en charge.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:609]:
```suggestion
> **Note :** Cette fonctionnalité apparaît dans Firefox 71, le seul navigateur pour le moment à la prendre en charge.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 12]
```diff
@@ -495,15 +598,17 @@ En définissant la propriété `display:grid` sur l'élément `box1`, il devient
...
 
-Selon la version actuelle de la spécifiction, il faudrait éditer l'exemple de grille imbriquée précédent et remplacer `grid-template-columns: repeat(3, 1fr)` en `grid-template-columns: subgrid`. La grille imbriquée utilisera alors la grille parente pour inscrire ses éléments dans le document.
+Selon la version actuelle de la spécifiction, il faudrait modifier l'exemple de grille imbriquée précédent et remplacer `grid-template-columns: repeat(3, 1fr)` en `grid-template-columns: subgrid`. La grille imbriquée utilisera alors la grille parente pour inscrire ses éléments dans le document.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:611]:
```suggestion
Selon la version actuelle de la spécification, il faudrait modifier l'exemple de grille imbriquée précédent et remplacer `grid-template-columns: repeat(3, 1fr)` en `grid-template-columns: subgrid`. La grille imbriquée utilisera alors la grille parente pour inscrire ses éléments dans le document.
```

J'en profite pour corriger une petite coquille :)

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 13]
```diff
@@ -518,7 +623,11 @@ Selon la version actuelle de la spécifiction, il faudrait éditer l'exemple de
 
 ## Superposer les éléments avec `z-index`
 
-Plusieurs éléments peuvent être placés dans la même cellule. Si nous retournons à notre exemple avec les items positionnés par numéros de ligne, nous pouvons modifier cela pour que deux items se chevauchent.
+Plusieurs éléments peuvent être placés dans la même cellule. Dans ce cas, il faut utiliser la propriété {{cssxref("z-index")}} pour contrôler l’ordre dans lequel les éléments se chevauchent.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:626]:
```suggestion
Plusieurs éléments peuvent être placés dans la même cellule. Dans ce cas, il faut utiliser la propriété {{cssxref("z-index")}} pour contrôler l'ordre dans lequel les éléments se chevauchent.
```

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md [hunk 14]
```diff
@@ -565,9 +676,9 @@ Plusieurs éléments peuvent être placés dans la même cellule. Si nous retour
 }
 ```
 
-{{ EmbedLiveSample('Superposer_les_éléments_avec_z-index', '230', '420') }}
+{{ EmbedLiveSample('Chevauchement_sans_z-index', '230', '460') }}
 
-L'élément `box2` est maintenant superposé avec `box1`, et comme il arrive après dans le code source il s'affiche par-dessus.
+L’élément `box2` chevauche maintenant `box1` ; il s’affiche par-dessus parce qu’il arrive après dans le code source.
```
Comment (cw118) [inline files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:681]:
```suggestion
L'élément `box2` chevauche maintenant `box1`&nbsp;; il s'affiche par-dessus parce qu'il arrive après dans le code source.
```

**Review by cw118 — CHANGES_REQUESTED** [review]
Thank you for the PR @pols12 ! Here are some minor comments/suggestions :)

> Does a tool exist to follow updates to a page and to see outdated translations?
Is each date update noted, in order to easier check what have changed since last update?

We are discussing and trying to plan a better way to show how up-to-date localized content is: https://github.com/orgs/mdn/discussions/129

> Do we have a tool to provide a diff for a given file between two dates?

I believe git and/or GitHub are already some great tools for seeing diffs.

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md (hunk 1 above)
Comment (pols12) [reply files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:21]: OK. Existe-t-il un précis des conventions typographiques pour la MDN en français ?

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md (hunk 4 above)
Comment (pols12) [reply files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:190]: Oui, j’aurais plutôt dû suggérer cela pour l’original en anglais.

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md (hunk 11 above)
Comment (pols12) [reply files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:609]: J’ai l’habitude d’utiliser les rectifications orthographiques de 1990, désolé.

File: files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md (hunk 9 above)
Comment (pols12) [reply files/fr/web/css/css_grid_layout/basic_concepts_of_grid_layout/index.md:540]: En effet, je me suis mélangé les pinceaux en regardant les commits un par un. Désolé.

**Review by cw118 — APPROVED** [review]
Merci beaucoup et félicitations pour cette contribution @pols12 ! Fusionnons tout ça :)

Voici le [guide stylistique de mozfr](https://github.com/mozfr/besogne/wiki/Guide-stylistique-pour-la-traduction), vous y trouverez la majorité des conventions utilisées sur MDN pour la locale française.

**Pull request merged**